huaweicloud/services/er/data_source_huaweicloud_er_flow_logs.go

```diff
@@ -0,0 +1,279 @@
+				Elem:        flowLogSchema(),
+				Computed:    true,
+				Description: "The list of the flow logs.",
+			},
+		},
+	}
+}
+
+func flowLogSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The ID of the flow log.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The name of the flow log.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The description of the flow log.",
+			},
+			"resource_type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The type of the flow log.",
+			},
+			"resource_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The ID of the attachment to which the flow log belongs.",
+			},
+			"log_group_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The ID of the log group to which the flow log belongs.",
+			},
+			"log_stream_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The ID of the log stream to which the flow log belongs.",
+			},
+			"log_store_type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The storage type of the flow log.",
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The creation time of the flow log.",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The latest update time of the flow log.",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the flow log.",
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "The switch of the flow log.",
+			},
+		},
+	}
+	return &sc
+}
+
+func dataSourceFlowLogsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// listFlowLog: Query the flow log list
+	listFlowLogsClient, err := cfg.ErV3Client(region)
+	if err != nil {
+		return diag.Errorf("error creating ER v3 client: %s", err)
+	}
+
+	listFlowLogsHttpUrl := "enterprise-router/{er_id}/flow-logs"
+	listFlowLogsPath := listFlowLogsClient.ResourceBaseURL() + listFlowLogsHttpUrl
+	listFlowLogsPath = strings.ReplaceAll(listFlowLogsPath, "{er_id}", d.Get("instance_id").(string))
+
+	listFlowLogsQueryParams := buildListFlowLogsQueryParams(d, cfg)
+	listFlowLogsPath += listFlowLogsQueryParams
+
+	listFlowLogsOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	listFlowLogsResp, err := listFlowLogsClient.Request("GET", listFlowLogsPath, &listFlowLogsOpt)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving flow logs")
```

Comment:
directly return the error will be fine
```suggestion
		return diag.Errorf("error retrieving flow logs: %s", err)
```

Reply:
done